deploy/mongodb/templates/clusterdefinition.yaml

```diff
@@ -33,10 +33,10 @@ spec:
           keys:
             - mongodb.conf
           defaultMode: 256
-        - name: mongodb-metrics-config
-          templateRef: mongodb-metrics-config
+        - name: mongodb-metrics-config-new
```

Comment:
why is refactoring name fix the bug?

Reply:
@sophon-zt could you explain?

Reply:
now the operation of kb upgrading will restart previous database pods, more info https://github.com/apecloud/kubeblocks/issues/3531
this behavior causes rendered config isn't  consistent with the needs of clusterdefinition, because the clusterdefinition is upgraded, but rendered config keeps remaining。
so renaming config-template is a cheap workaround, but not the fundamental solution

Reply:
that's a hack then, is it possible to get a real fix?

Reply:
the radical reason is restarting pods when upgrading kubeblocks, and this pr will mitigate the restarting impact